civiform/civiform
# Move focus to input field when a single question on an application block has errors

### Description

Per our accessibility advisors' recommendation, moving focus to the first field with an error when a single question on an application block has errors.  This directs the screen reader directly to the field that needs to be updated.  The screen reader then reads the question and the error message.  When there are multiple questions with errors, focus is not redirected.  Rather, we have kept the existing behavior in which the screen reader gives a vocal alert of there being multiple errors on the page.

## Release notes

Per our accessibility advisors' recommendation, moving focus to the first field with an error when a single question on an application block has errors.  This directs the screen reader directly to the field that needs to be updated.  The screen reader then reads the question and the error message.  When there are multiple questions with errors, focus is not redirected.  Rather, we have kept the existing behavior in which the screen reader gives a vocal alert of there being multiple errors on the page.

### Checklist

#### General

Read the full guidelines for PRs [here](https://docs.civiform.us/contributor-guide/developer-guide/technical-contribution-guide#guidelines-for-pull-requests)


#### User visible changes


### Instructions for manual testing


1. Ensure a program has a required question.
2. Turn on a screen reader.
3. Begin filling out the application and hit "Next" without filling out the required question.
4. Notice that the screen reader goes directly to the missing field.

### Issue(s) this completes

Fixes #<[3287](https://github.com/civiform/civiform/issues/3287)>;
